gui/editor/editor.cpp

```diff
@@ -1001,6 +1001,40 @@ void NumeReEditor::OnChar( wxStyledTextEvent& event )
     if (sAutoCompList.length())
         AutoCompShow(lenEntered, sAutoCompList);
 
+    // if line indicator setting is active, provide autowrapping for comments
+    bool bLineIndicatorSet = m_options->getSetting(SETTING_B_LINELENGTH).active();
+    const int iMaxLineLength = 100;     // member from codeanalyzer.hpp private MAXLINESOFCODE
+    const int iColumnPos = GetColumn(currentPos);
+
+    if (bLineIndicatorSet && (iColumnPos > iMaxLineLength))
+    {
+        int indentWidth = GetLineIndentation(currentLine);
+
+        if (GetStyleAt(currentPos) == wxSTC_NSCR_COMMENT_LINE)
+        {
+            InsertText(currentPos-1, "\r\n## ");
+            GotoPos(currentPos+5);    		  // carret to pos ## |
+        }
+        else if (GetStyleAt(currentPos) == wxSTC_NSCR_COMMENT_BLOCK)
+        {
+            InsertText(currentPos-1, "\r\n * ");
+            GotoPos(currentPos+5);    		  // carret to pos * |
+        }
+        else if (GetStyleAt(currentPos) == wxSTC_NSCR_DOCCOMMENT_LINE)
+        {
+            InsertText(currentPos-1, "\r\n##! ");
+            GotoPos(currentPos+6);    		  // carret to pos ##! |
+        }
+        else if (GetStyleAt(currentPos) == wxSTC_NSCR_DOCCOMMENT_BLOCK)
+        {
+            InsertText(currentPos-1, "\r\n#*! ");
+            GotoPos(currentPos+6);    		  // carret to pos #*! |
+        }

```

Comment:
This block can be the same as the one for the `wxSTC_NSCR_COMMENT_BLOCK`, because the additional `#*!` is not necessary (you are already within a block right now).

Reply:
remove the case differentiation